entgql/pagination.go

```diff
@@ -218,16 +218,32 @@ func multiPredicate[T any](cursor *Cursor[T], opts *MultiCursorsOptions) (func(*
 	return func(s *sql.Selector) {
 		// Given the following terms: x DESC, y ASC, etc. The following predicate will be
 		// generated: (x < x1 OR (x = x1 AND y > y1) OR (x = x1 AND y = y1 AND id > last)).
+
+		// getColumnNameForField gets the name for the term and considers non-ambigous matching of
+		// terms that may be joined instead of a column on the table.
+		getColumnNameForField := func(field string) string  {
+			// The predicate function is executed on query generation time.
+			column := s.C(field)
+			// If there is a non-ambiguous match, we use it. That is because
+			// some order terms may append joined information to query selection.
+			if matches := s.FindSelection(field); len(matches) == 1 {
+				column = matches[0]
+			}
+			return column
+		}
```

Comment:
using `getColumnNameForField` allows to order by joined data rather than only columns on the table